### PR TITLE
Fix concurrent map iteration & write issue

### PR DESCRIPTION
**Pull request by Blaxon**
The issue was introduced in `func (l *fileStore) Remove(addr string) error`
https://github.com/qiniu/goc/blob/master/pkg/cover/store.go#L107

when we set the memory server list back to file. 
the list `l.memoryStore.GetAll() -> services` passed into `func (l *fileStore) Set(services map[string][]string) error`
in [L175](https://github.com/qiniu/goc/blob/master/pkg/cover/store.go#L175) we sign the `services` directly to `l.memoryStore.servicesMap`, without copying it.

while in [L186](https://github.com/qiniu/goc/blob/master/pkg/cover/store.go#L186) we are still using this map to do the iteration without lock. (by right should use `memoryStore.mu`.

The issue will happen when there is another goroutine tries to remove the server from memoryStore. Map will clash.

### CODE REVIEW

**Comment by Blaxon** [comment]
Solve issue: https://github.com/qiniu/goc/issues/324

File: pkg/cover/store_test.go [hunk 1]
```diff
@@ -130,3 +130,22 @@ func TestFileStoreRemove(t *testing.T) {
 	err = store.Remove("http")
 	assert.Error(t, err, fmt.Errorf("no service found"))
 }
+
+// verify issue fix https://github.com/golang/go/issues/56552
+func TestConcurrentRemoval(t *testing.T) {
```
Comment (CarlJi) [inline pkg/cover/store_test.go:135]: it's better to add more direct verification as [this](https://stackoverflow.com/questions/31595791/how-to-test-panics)

Comment (Blaxon) [reply pkg/cover/store_test.go:135]: Hi @CarlJi according to the document we can use `recover()` to catch panic, but our issue is `Fatal Error` which cannot recover. From my understanding `Fatal Error` shouldn't exists in a healthy program. So i'm not sure do we still need this case? Thanks!

Comment (Blaxon) [reply pkg/cover/store_test.go:135]: Hi @CarlJi I think our issue it `Fatal Error` which is not recoverable according to[this](https://stackoverflow.com/a/39289246). 
But I just realize the logic for `fileStore.Set` will inevitably causing data inconsistency as it separate 2 storage sync in 2 lockers. as like this case, even I wait for all the goroutine finished, the serviceMap might still have data not been cleared. 
I'll check it further tmr!

**Comment by Blaxon** [comment]
Hi @CarlJi , I just made a new commit 
1. to fix the fileStore.Remove data inconsistency issue,
2. and update the unit test to be verifiable.
Pls help check, Thanks!

**Pull request merged**